config/azure.go

```diff
@@ -81,23 +80,18 @@ func (c *Config) azureEnsureWorkspaceUrl(ctx context.Context, ahr azureHostResol
 	}
 	// azure resource ID can also be used in lieu of host by some of the clients, like Terraform
 	management := ahr.tokenSourceFor(ctx, c, env, env.ResourceManagerEndpoint)
-	resourceManager := oauth2.NewClient(ctx, management)
-	resp, err := resourceManager.Get(env.ResourceManagerEndpoint + c.AzureResourceID + "?api-version=2018-04-01")
-	if err != nil {
-		return fmt.Errorf("cannot resolve workspace: %w", err)
-	}
-	raw, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return fmt.Errorf("cannot read: %w", err)
-	}
 	var workspaceMetadata struct {
 		Properties struct {
 			WorkspaceURL string `json:"workspaceUrl"`
 		} `json:"properties"`
 	}
-	err = json.Unmarshal(raw, &workspaceMetadata)
+	requestURL := env.ResourceManagerEndpoint + c.AzureResourceID + "?api-version=2018-04-01"
+	err = httpclient.DefaultClient.Do(ctx, "GET", requestURL,
+		httpclient.WithUnmarshal(&workspaceMetadata),
+		httpclient.WithTokenSource(management),
+	)
 	if err != nil {
-		return fmt.Errorf("cannot unmarshal: %w", err)
+		return fmt.Errorf("resolve workspace: %w", err)
```

Comment:
Since we are throwing a different error will this break tests (if any) that checks for this substring. Also should the error be `cannot resolve workspace` instead?

Reply:
TODO